### PR TITLE
Migrate to move v6 and u16, u32, u256 support

### PR DESCRIPTION
**Pull request by oxade**
After ensuring Move v6 types are backward compatible and not breaking existing Sui, we're now ready to migrate to Move v6.
This enables newer features including new integer support like u16, u32, u256.
Follow up PRs will flesh out new integer support for Typescript modules & apps if needed.

### CODE REVIEW

**Comment by oxade** [comment]
@damirka I could use some feedback on the BCS TS module please.

File: crates/sui-framework/sources/math.move
Comment (amnn) [inline crates/sui-framework/sources/math.move:93]: Per the offline discussion -- we should go down the route of a different math module per integer width.  This can be done in a different PR though (probably it makes sense to split this change out anyway).

File: crates/sui-framework/tests/math_tests.move
Comment (amnn) [inline crates/sui-framework/tests/math_tests.move:29]: Should have a test that makes sure this works for values that fit in u128 but not u64.

**Review by amnn — COMMENTED** [review]
Overall looks good to me, but I'm not personally familiar with this process of updating our move integration, so holding off on the stamp for that reason.

File: sdk/typescript/src/types/sui-bcs.ts
Comment (damirka) [reply sdk/typescript/src/types/sui-bcs.ts:10]: Thanks! I expected this to happen and will use "theirs".

File: crates/sui-framework/sources/math.move
Comment (oxade) [reply crates/sui-framework/sources/math.move:93]: Yes I think separation is a diff PR. This case here was just added to illustrate usage of u256.

File: crates/sui-framework/tests/math_tests.move
Comment (oxade) [reply crates/sui-framework/tests/math_tests.move:29]: Right. Didn't want to mix too many unrelated things in this PR. Fleshing out `math.move` should happen separately.

File: crates/sui-json/src/lib.rs
Comment (oxade) [reply crates/sui-json/src/lib.rs:622]: There probably is. I can try to benchmark this and follow up with a tweak if needed.

**Review by damirka — APPROVED** [review]
Overall looks good to me. Multiple vector definitions will go away once #5427 is landed.

**Pull request merged**
